x-pack/plugins/data_enhanced/server/search/es_search_strategy.ts

```diff
@@ -45,9 +51,24 @@ export const enhancedEsSearchStrategyProvider = (
     const defaultParams = getDefaultSearchParams(config);
     const params = { ...defaultParams, ...request.params };
 
-    return request.indexType === 'rollup'
-      ? rollupSearch(caller, { ...request, params }, options)
-      : asyncSearch(caller, { ...request, params }, options);
+    try {
+      const response =
+        request.indexType === 'rollup'
+          ? await rollupSearch(caller, { ...request, params }, options)
+          : await asyncSearch(caller, { ...request, params }, options);
+
+      if (
+        usage &&
+        (!isAsyncSearchResponse<any>(response) || (!response.is_partial && !response.is_running))
+      ) {
+        usage.trackSuccess(response.rawResponse.took);
```

Comment:
You also need to add an else clause with 

```
if (response.is_partial && !response.is_running) {
    usage.trackError();
}
```

And adding some information to that failure would be great.

Reply:
Agreed, we'll have to add additional metadata in a subsequent PR but I cant add this error case.